### ISSUE TITLE
Error in DETR Loss

### ISSUE THREAD

**Issue by ananya-singhh** [tensorflow/models#10926]
# Prerequisites

Please answer the following questions for yourself before submitting an issue.
- [x] I am reporting the issue to the correct repository. (Model Garden official or research directory)
- [x] I checked to make sure that this issue has not been filed already.

## 1. The entire URL of the file you are using

https://github.com/tensorflow/models/official/projects/detr/tasks/detection.py

## 2. Describe the bug

The TensorFlow implementation of DETR uses Box loss and GIoU loss. Focal loss and dice loss are not mentioned anywhere in DETR.

## 3. Steps to reproduce

The loss functions used in the TensorFlow implementation of DETR can be found [here](https://github.com/tensorflow/models/blob/58261d1d92d507cb4e656984411cc0acbb8234bc/official/projects/detr/tasks/detection.py#L163).

When searching for "detr focal" or "detr dice" or looking through all the files in the project for DETR, one can confirm no mention of focal and dice loss.

## 4. Expected behavior

In the paper End-to-End Object Detection with Transformers and associated [repository](https://github.com/facebookresearch/detr), the official PyTorch DETR implementation uses [focal loss](https://github.com/facebookresearch/detr/blob/8a144f83a287f4d3fece4acdf073f387c5af387d/models/segmentation.py#L190) and [dice loss](https://github.com/facebookresearch/detr/blob/8a144f83a287f4d3fece4acdf073f387c5af387d/models/segmentation.py#L172).

## 5. Additional context

N/A

## 6. System information

N/A

**Comment by laxmareddyp** [comment]
Hi @fyangf ,

Could you please look into this once in your free time and there are couple of [PR's](https://github.com/keras-team/keras-cv/pull/968) opened related to dice loss, yet to be merged.

Thanks.

**Comment by fyangf** [comment]
The `sigmoid_focal_loss` and `dice_loss` seen from the official implementation is for masks: https://github.com/facebookresearch/detr/blob/8a144f83a287f4d3fece4acdf073f387c5af387d/models/detr.py#L336-L338. Since we only reproduced detection results but not the instance segmentation results yet, we have not added support for these losses.

**Comment by ananya-singhh** [comment]
I see, that makes sense!
